NOTICE

```diff
@@ -0,0 +1,26 @@
+###################################################################################
+
+The following modifications follow Apache License 2.0 from shiguredo.
+
```

Comment:
So far, we have used three patches from Shiguredo, I think we will follow more Shiguredo patches.